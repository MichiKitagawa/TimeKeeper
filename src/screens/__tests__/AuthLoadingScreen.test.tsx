import React from 'react';
import { render, screen } from '@testing-library/react-native';
import AuthLoadingScreen from '../AuthLoadingScreen';

describe('AuthLoadingScreen', () => {
  it('正しくレンダリングされ、ActivityIndicatorとLoadingテキストが表示される', () => {
    render(<AuthLoadingScreen />);

    // ActivityIndicatorの存在確認 (testIDなどで特定できるとより良い)
    // 今回はActivityIndicatorコンポーネントが一つだけ表示されることを期待
    const activityIndicator = screen.UNSAFE_getByType('ActivityIndicator'); // RNのActivityIndicatorコンポーネントを直接指定
    expect(activityIndicator).toBeTruthy();
    expect(activityIndicator.props.size).toBe('large');

    // "Loading..." テキストの存在確認
    expect(screen.getByText('Loading...')).toBeTruthy();
  });

  it('スナップショットテスト', () => {
    const tree = render(<AuthLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
}); 