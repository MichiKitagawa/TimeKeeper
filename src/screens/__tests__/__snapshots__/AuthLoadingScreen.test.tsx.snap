// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLoadingScreen スナップショットテスト 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    size="large"
  />
  <Text>
    Loading...
  </Text>
</View>
`;
